### ISSUE TITLE
Core/Chapter-org: Why are you (not) motivated?

### ISSUE THREAD

**Issue by martinheidegger** [nodeschool/organizers#370]
I am not sure of how to ask this question to @nodeschool/core and @nodeschool/chapter-organizers and I am sure the answers will be diverse but:

## Are you steadily working on NodeSchool?
_(Hold events, improve the homepage/github issues, write tutorials, etc.)_

### `if (thingsGoingWell) {`
- What advice would you give to people that are not motivated?
- What keeps you motivated?
- Did you get over a uncomfortable period?

### `} else {`
- What is missing?
- Can we make your situation better?
- What would you expect us to help you with?
- DId something piss you off?

### `}`

_If your thoughts are nsfw. please feel free to write a private message to my github email address._

**Comment by RichardLitt** [comment]
@martinheidegger Can we have some context on why you're asking this? Should I respond here, using this template?

**Comment by martinheidegger** [comment]
@RichardLitt I am asking this because I think I have a gut feeling about this but I would love to have some real information. You can respond however you feel like.

**Comment by denysdovhan** [comment]
In my chapter everything is going well. But mostly we have problems with sponsors and money.

What keeps me motivated? It may seem quite naive, but I just love that people like it. They ask me when will be next event, what can they help.

Uncomfortable period: here it is. We have to move our chapter to the new location. Currently we don't know where next, but are looking for a good place. Also on the last event we had not so much members. Maybe because workshoppers don't provide a complete learning path (I heard that from peoples).

**Comment by andrenventer** [comment]
Hi Martin.if (thingsGoingWell) {

   - we are stuck

} else {

   - People are missing.
   - You cannot help us find people, we have a small town. The only way I
   can se it work for us is on-line, so location is not the defining boundary.
   - What would you expect us to help you with? We enjoy the materials -
   Big thanks for that!
   - DId something piss you off? No

}

**Comment by RichardLitt** [comment]
## Are you steadily working on NodeSchool?

No. I helped organize events a couple of years ago - in SF, and then I hosted two in SouthEast Asia while traveling - but I haven't prioritized it recently. I do read most of the issues for the nodeschool organization on GitHub, but that's more because I subscribe than anything else, and I don't help all that often. 

### `if (thingsGoingWell) {`
- What advice would you give to people that are not motivated?
- What keeps you motivated?
- Did you get over a uncomfortable period?

### `} else {`
- What is missing?

I don't have a community where I am right now that I am actively involved in that is doing NodeSchools. I live in Cambridge, MA - I know that there are some node events going on, but I'm not really involved. This is more my fault than anything else. I run Somerville-JS (which means choosing a place to go hang out once a month), and go to BostonJS, but that's it for now. I could get more involved, but I don't know how much bandwidth I have. 

I also, frankly, just don't interact that much with Node right now. I don't really work on webapps anymore, and I feel that the small CLI stuff I do and the random JS packages aren't really all that relevant to the greater node community. I don't see running a NodeSchool as useful, for me or anyone else, because most Noders want to build webapps, and I suck at that right now. 

- Can we make your situation better?

Not really? I can reach out to Nodeschool people here if I really wanted to host and run one. At the moment, I am a remote worker, so I don't have a space or anything, and the hassle of trying to arrange that might be too much for me. 

On the GitHub organization, no. It seems to be pretty well handled at the moment. There could be a doc on how to use NodeSchool bot, but I just don't get involved all that much right now (again, priorities and time). 

- What would you expect us to help you with?

Nothing. I'm expected to pull some weight if I volunteer, and I feel that I haven't. I'm not upset about this - if I wanted to, I could. 

- Did something piss you off?

I see a lot of drama on Twitter, and at various JS conferences, and I don't like getting involved. I've felt a lot of judgement over the past few years because of my particular set of perceived attributes I can't really control (ethnicity, sex identity, sexual preference, background, nationality, and education, among others), and that judgement has recently made me feel very unwelcome in several venues. It has been easier for me to back away from contributing to large conferences or places where I feel unwelcome than it would be for me to have an open discussion about it, especially against, in at least two instances, openly hostile participants. 

So, did something piss me off? Not really. I'd say I've just been backing away from confrontation. This isn't just about being a white male, either; I'm not sure that _me_ running a NodeSchool in third world countries really does benefit the local population in a non-colonial way, and I'm not going to run any more there until I think about the process more in depth. (Of course, I don't have this issue in Boston). 

### `}`

**Comment by NorthDecoder** [comment]
Martin, thanks for asking!

In 2014 a friend recommended that I teach to share 
some of the stuff that I know.  It seemed like a 
bad idea at first, but then I thought that if the 
teaching only included people that were really 
interested and there were no mandatory deadlines, 
then I might be interested.  So I searched the web 
for local interest groups and found that they are 
now called meetups.  In my search to find a common 
language, I was surprised to see how ubiquitous 
Javascript had become, that there was no meetup 
here in Tacoma, and that I had around 15 people 
attend the first meetup. I have tried to host a 
meetup one Saturday per month since then with
varying numbers of attendees.

We have switched between browser and node topics. 
At each event I ask what type of laptop each 
person is working with and what they have at home 
and at the office.   The type of development 
system is really important for working with 
nodejs.  A few attendees arrive with brand new 
company paid equipment, but many are working with 
older equipment that they are paying for 
themselves. The portable pads are so minimal we 
have had trouble with keys not mapping correctly 
on the command line.  A few real hackers arrive 
with linux installed on an old laptop that has 
extra batteries wired to the outside.  To be clear 
the batteries do not make one a better node 
programmer, but they do help improve the hacker 
status.  To be relatable to attendees I have 
avoided working with the latest hardware upgrade.

That means working with a hand-me-down macbook
darwin version 10.8.  I have had trouble getting new
versions of node with any package manager, so
have used the binary downloads.  I am writing an
[article](http://vividventures.biz/d/?q=installing_nodejs_without_package_manager) about how to do that correctly,
but am unsure that the directory ownership 
definition is correct, so I would like feedback
on that; or any other issue with the article.

On the other hand, the world does not care what
works on my development machine as long as the
public app works, and that it is properly backed
and controlled with scm.  So we include detailed
discussion about git and how it relates to our
example project, and we have experimented with
deploying to several infrastructure-as-a-service 
companies.

Copying the tradition of some other meetups, I
have scheduled after work no-host Taco Thursdays
this summer as a way for newcomers to attend and
decide if the group is friendly and worthwhile.
The best part is that we get to converse with
people who bring all sorts of knowledge to the
table.

Being relatable to everyone is difficult.  Each
person that attends has a unique situation, skill
set, end results goal, interest level etc.

There are a couple of other related meetups, Big 
Data and Tacoma Dev Ops that I have taken 
responsibility for leadership and those take time 
away from my ability to focus only on nodeschool.
Oh, and most importantly, I have family, home and
friends that need me too!

**Comment by llkats** [comment]
I co-organize NodeSchool San Francisco together with @jedireza. We meet monthly at the public event space at Mozilla SF.

SF is situated in a geographic area of great need and demand for code education. I know a few other chapters that have a similar demand, but it seems like not many do. Additionally, we have a great sponsor in Mozilla, who has hosted us monthly for over a year. We're absurdly lucky in those regards. 

Since we don't have many logistics to handle from month-to-month, we've been able to explore things like having focused sessions (having a volunteer present on a topic or work through a workshopper with a group all together) or having mentor hack nights to work on the workshoppers themselves. I think the variety has helped us not get burned out or de-motivated.

### `if (thingsGoingWell) {`

- What advice would you give to people that are not motivated?
	- Don't feel like you need to have events every month. If people aren't motivated or interested, wait for a month or three until demand or interest builds again, and mentors are more motivated. IIRC, SF has skipped holding events in December because the holidays are too stressful and busy for everyone.
	- When you do have events, work those social media accounts and network to spread the word and drive interest. Stress the "free" and "casual" aspects. Reach out to other programming meetups and events and teaching groups to see if they will promote your event as well.
	- If there's a nearby chapter, reach out and see if you can co-host an event. Every now and then, we "field trip" over to NodeSchool Oakland, or they co-host with us in SF. It gets a different audience and set of mentors, shakes up the routine a bit, and fosters a sense of regional community.
	- Have coffee, drinks, or ice cream with your mentors after the event. The decompression is nice and it's a nice reward after a big day of expending social energy. I probably look forward to that part of the day most of all. :beers: :blush: 
- What keeps you motivated?
	- Seeing our attendees not only learn to code, but also grow networks and join the Node community, keeps me going. We try to encourage attendees to meet and work with each other during our events. People make friends at SF. That's something that I think is really special.
- Did you get over a uncomfortable period?
	- When we were still moving monthly between venues, that was a lot of logistics to handle. I really feel for chapters that have to do that all the time, it's a lot of energy and planning to take on. When we settled at Moz, it was a bit of a relief because we could focus on other things.

### `} else {`

- What is missing?
	- There's not a sense of forward leadership or direction with NodeSchool. It feels a bit stalled with regards to developing the workshoppers, and the quality of those modules has degraded with time. This has especially been apparently when we've tried to hold focused sessions with them (notable the `how-to-npm` and the react workshoppers). I wonder about how to ensure the workshoppers stay high-quality, how to make them more accessible to improve, and what is the overall direction for NodeSchool learning materials.
	- We do have a problem with ticket no-shows, which is pretty frustrating and deflating. I'm not sure anyone can really help with that.
- Can we make your situation better?
	- Having a little more understanding of the overall direction of NodeSchool would be nice, so we could organize and contribute to the overall curriculum.
- What would you expect us to help you with?
	- Improving workshoppers (that's the theme of my feedback apparently!)
- DId something piss you off?
	- Nope! Not pissed in the slightest. :)

### `}`

Hope this helps!

**Comment by martinheidegger** [comment]
@denysdovhan @andrenventer @RichardLitt @NorthDecoder @llkats I am keeping your responses here close to my heart, thank you for taking the time. I leave this issue open for a little longer in hope to hear more voices from @nodeschool/core @nodeschool/chapter-organizers

**Comment by Fauntleroy** [comment]
### Well

#### What advice would you give to people that are not motivated?
- If you're having difficulty maintaining motivation (and you're the only organizer) it might be a good idea to recruit a co-organizer, or develop a team. Spreading the workload, getting new ideas, and feeling the energy of others around you can be very helpful.

#### What keeps you motivated?
- At the end of every event around 10 or so people will come up to me personally and thank me for running the event. Our attendees really seem to love the event, and that's a great help motivation-wise.
- Mentors always seem happy to help and offer words of encouragement near-constantly.
- We order some hella good pizza 🍕 for these events.

#### Did you get over a uncomfortable period?
Yes—when I first inherited the meetup group, we hadn't had a meetup for an extremely long time. I basically needed to get everything set up from zero, but luckily I had a great deal of help from @llkats, @jedireza, and Isaac from npm. Once I got things moving we became able to have an event every month, and the regularity has helped things a great deal.

### Not Well

#### What is missing?
A lil' more organization—I know what the SF chapter is up to because I speak to their organizers often, but I honestly have no idea what any of the other chapters are up to. Maybe lots of communication happens in the Gitter/issues, but I feel like I have no idea what the other chapters are up to.

#### Can we make your situation better?
Many NodeSchool workshoppers are in a state of disrepair. It'd be lovely if there were a concerted effort to go through and modernize workshoppers. We've had issues with things like text coloring being unreadable in CLIs, workshoppers not working in Windows, and some workshoppers not working at all.

#### What would you expect us to help you with?
Honestly I'm not sure... I've never thought of getting help from y'all. Aside from a few interactions at the beginning, I haven't talked to NodeSchool at all.

#### Did something piss you off?
Nothing directly related to NodeSchool.

**Comment by jaybobo** [comment]
I co-organize in Central Ohio with the lovely people of [PairColumbus](http://paircolumbus.org). We're a language agnostic group focused on pairing and mentoring so we don't have the same issues that other groups might have as nodeschool is just one component of many. Monthly attendance is 30-40 people.

`if (thingsGoingWell) {`

#### What advice would you give to people that are not motivated?
Be flexible. We widely vary from month to month, sometimes we work on nodeschool tutorials but we've also done things like...

- An intro to OWASP with a beginner-friendly CTF
- A mini-hackathon where the goal was to do fun stuff with an API of your choice
- Cryptoparty
- A group session where we hacked on the same open source MeteorJS app
- Automated testing & TDD kata's
- Prepping for technical interviews

#### What keeps you motivated?
Knowing that we're making people's lives better. We've had so many people new and experienced devs use our curriculum and that of nodeschool to get jobs or transition into different roles.

#### Did you get over a uncomfortable period?
No, not really. We embrace when things go wrong and pivot during the session or before our next meetup.

`} else {`

#### What is missing?
Not much just encourage others' creativity. There's opportunities to grow in the direction of wherever node and ECMAscript are being used across the industry.

#### Can we make your situation better?
Some workshoppers could be better, but other than that its great!

#### What would you expect us to help you with?
Don't stop. Keep fighting the good fight. 👯 

#### DId something piss you off?
Nope not yet.


`}`

**Comment by tdd** [comment]
Hey all, sorry I'm late to this party. Vacation, baby boy and stuff 😉 

So here are my 2 cents.

I run [NodeSchool Paris](https://www.meetup.com/fr-FR/NodeSchool-Paris/), which tries to stick very close to the original NS format: full-day free event for anyone to try their hand at any NS workshop in a friendly setting, with mentors, etc. We're trying to have one every 6-7 weeks, mostly because this is as short a timespan between two absent Saturdays as my wife will let me get away with 😀 

We're blessed with an easy road for every session:

1. Mozilla Paris hosts us for free, providing a kick-ass venue with great beverage choices (in quantity, quality and variety: a lot is non-alcoholic, too, which is awesome).
2. My own company sponsors the catering as a matter of course; sometimes additional sponsors help split up the bill (we provide high-end, diverse catering for 60 pax which comes in at a fixed €600 VAT-excl. every time).

> What advice would you give to people that are not motivated?

Don't try and make it perfect; don't try and make it happen at too high a frequency; it's okay to take breaks; it's okay to delegate to fellow event runners; it's ok to resort to non-experts for mentoring: they just need to grasp the topics better, or at least no worse, than the other attendees 😉   Above all, make sure your event is *social*: it's at least as much about meeting the local community, incl. complete rookies/beginners, as it is about learning JS/Node.

> What keeps you motivated?

The enjoyment of attendees, and the tons of ah-ha moments and eye-sparks we see in them across sessions. We're giving a lot of people (despite a strict 60-pax cap and not too many sessions) a great headstart / leg-up to JS, Node and other stuff, and that feels amazing.

> Did you get over a uncomfortable period?

Fortunately, no.

> What is missing?

- A more homogeneous platform / subtrate across workshoppers. Esp. wrt translation.
- More maintenance on workshoppers so they work on the latest Nodes (and on Win!) w/o glitches
- More FR translations (our bad; Jérémie and I worked hard in early 2015 to g11n as much as we could, but we dropped that ball after a few months).
- Better communication channels between chapters in nearby timezones / locations, to help build common events / dates / crossovers.

> Can we make your situation better?

- See above.
- Also, tons of NS chapters are run through sites like Meetup.com / EventBrite. Providing an easy way for the main NS site to get their next events and info from the public API for these sites would be great; less duplication/maintenance effort for organizers.

Best,

**Comment by gangstead** [comment]
I co-organize @nodeschool/dallas with @samiconductor @prochafilho @michaeledavis . We've met monthly for 20 months now.  It's going well but it's getting harder as time goes on.
### `if (thingsGoingWell) {`

- **What advice would you give to people that are not motivated?**
 - Find a format that works, as simple as possible, and keep to it.  Having a simple format that doesn't change keeps the monthly amount of pre-meetup overhead down to less than an hour, plus doing the module ahead of time.
- **What keeps you motivated?**
 - We've had steady attendance and usually get around 20-30 people with about half new comers.  It's been a great way to meet new people and the response is always positive.
- **Did you get over a uncomfortable period?**
 - We didn't really have an uncomfortable period... or maybe we're due for one?

### `} else {`

- **What is missing?**
 - Modules for all skill levels.  There are a couple very entry level ones, a ton of "scratch the surface" introductory modules and hardly any more advanced modules.
- **Can we make your situation better?**
  - *Maintain workshoppers*.  Maybe they need to be transferred to the nodeschool organization (on github and nodeschool be an owner on npm) before being featured on the website?  Core workshoppers like `learnyounode` should not have 100+ open issues and 10+ open pull requests.  There are a bunch of workshoppers out there that need basic maintenance but the creators have moved on and there's not always a clear fork that should be followed.  Then there are workshoppers that have a good start, maybe 2-3 exercises, but need more written before they are really ready.
- **What would you expect us to help you with?**
- **Did something piss you off?**
 - Not really, but there have been workshoppers we've wanted to feature but we can't get them to work in time for the meetup.

### `}`

To sum up: Nodeschool/Dallas has been going well but we need a way to help keep workshoppers up to date.  Opening PRs on repos that no one is maintaining is very frustrating.

**Comment by ThibWeb** [comment]
## Are you steadily working on NodeSchool?

Yup! I've been contributing to workshoppers now and then, now starting a chapter in the city I moved to.

### `if (thingsGoingWell) {`

I think things are going well because I've set my expectations according to what I believe NodeSchool to be – volunteer run by people in their spare time, without much financial backing, open-source, "grass roots". That works well if/when people dedicate their motivation and energy to it, which is not always the case, nor everyone's cup of tea.

### `} else {`

My ongoing concern with NodeSchool is that there isn't enough focus on the quality of the workshoppers. For newcomers, this means getting frustrated when a `npm install -g <workshopper-name>` fails on their system, not understanding the instructions of an exercice, not understanding the error messages.

**I think what's missing is some quality standards for workshoppers to be featured on http://nodeschool.io/.** I imagine this is nothing new though. Here are things I can think of off the top of my head:

- Green CI tests for all exercices with their official solutions, in Node 0.12 / LTS and latest, on Windows / Linux, in every translation of the workshopper.
- Translations possible by just adding extra language definitions (it's ok not to have any, but adding one shouldn't require refactoring)
- Well-defined conventions/standards for what the README / presentation page of each workshopper should contain.

I'll stop there. I think the NodeSchool core team is in a good position to address this by setting those standards and enforcing them, but ultimately it's up to the workshopper authors and contributors to make it happen.

### `}`

Hope that's useful.

**Comment by pcuci** [comment]
Can we fork these workshoppers in NodeSchool org anyway and let the
original maintainers upstream pull updates as they want, and viceversa?

An officially maintained fork for each workshopper

Happy to help out, having our first NodeSchool event in Montreal next week,
and honestly, I'm scared about ```npm install -g <workshopper>``` not
working for people, and failing on first impressions

**Comment by martinheidegger** [comment]
Okay, first off: Thanks everyone. It has been great that you took the time to answer here.
I am trying to pull out some time somewhere, somehow to write a wiki entry about my POV on all this.

Regarding the workshoppers: There is a "http://github.com/workshopper" organization that exists in order to not have even more repositories directly in the nodeschool. It contains many workshoppers. This organization is unfortunately no organization since there is no structure in it (everybody just somehow works as she pleases). It is possible - though I am not sure if its a good  idea - to move at least those workshoppers to nodeschool.

**Comment by martinheidegger** [comment]
I collected my thoughts and published them on Medium: https://medium.com/@leichtgewicht/the-missing-hands-at-nodeschool-8999a90d33d1#.wzsbfg41b

**Comment by iancrowther** [comment]
I'm pretty late to this but...

I've been reading through comments and articles (thanks @martinheidegger) and can very much relate the feedback regarding the overall leadership of Nodeschool and the ongoing maintenance of the workshoppers.

I've been digesting this with a couple of fellow organisers and am concluding that the project could be experiencing some "growing pains".

I am a big fan of the organisational structure in place and love the fact that it's been so successful without any form of formal governance. However, I do think we could (with care) look to make a lightweight plan to give everyone a steer.

I know I'm struggling to understand the projects objectives and think this could be an interesting place to start.

Any thoughts? Is it worth a PR?

**Comment by martinheidegger** [comment]
I published my account on medium if you are interested: https://medium.com/@leichtgewicht/introspection-of-an-open-source-effort-2b9577272898#.wbe17cm3f
